compiler/rustc_codegen_ssa/src/mir/operand.rs

```diff
@@ -545,67 +545,109 @@ impl<'a, 'tcx, V: CodegenObject> OperandRef<'tcx, V> {
         };
         OperandRef { val, layout }
     }
+
+    pub(crate) fn supports_builder(layout: TyAndLayout<'tcx>) -> bool {
+        match layout.backend_repr {
+            BackendRepr::Memory { .. } if layout.is_zst() => true,
+            BackendRepr::Scalar(_) | BackendRepr::ScalarPair(_, _) => true,
+            _ => false,
+        }
+    }
```

Comment:
is this builder used in so many places that this is simpler than just returning `Result`? or `Option`, I suppose